frontend/src/metabase/common/components/EllipsifiedPath/EllipsifiedPath.tsx

```diff
@@ -0,0 +1,52 @@
+import { useAreAnyTruncated } from "metabase/hooks/use-is-truncated";
+import { Tooltip } from "metabase/ui";
+
+import S from "./EllipsifiedPath.module.css";
+
+type EllipsifiedPathProps = { items: string[]; tooltip: string };
+
+/**
+ * Displays a path such as "Collection / Subcollection / Subsubcollection /
+ * Parent Collection".
+ *
+ * If the path is too long to fit, some items may be truncated, like this:
+ * "Collection / Subcollec... / Subsub... / Parent Collection".
+ *
+ * The first and last item are never truncated.
+ *
+ * A tooltip is shown if any items are truncated.
+ */
+export const EllipsifiedPath = ({ items, tooltip }: EllipsifiedPathProps) => {
+  const { areAnyTruncated, ref } = useAreAnyTruncated<HTMLDivElement>();
+
+  return (
+    <Tooltip label={tooltip} disabled={!areAnyTruncated} multiline maw="20rem">
+      <div className={S.path}>
+        {items.length > 1 && (
+          <div className={S.dots}>
+            … <div className={S.slash}>/</div>
+          </div>
+        )}
```

Comment:
if we get to a narrow screen width where this block cuts off all leading paths, but the final collection name doesn't get truncated, we don't get a tooltip. this is a minor thing, I don't think it should block this PR, and I don't think we should increase the complexity of this component to handle it, but wanted to call it out.

![Screen Shot 2024-08-14 at 9 44 01 AM](https://github.com/user-attachments/assets/fb1cf10a-d5a0-4002-bba3-c69817e523d1)